src/freenet/client/filter/HTMLFilter.java

```diff
@@ -2616,7 +2619,11 @@ void processStyle(HTMLParseContext pc) {
 			String href = getHashString(h, "href");
 			if (href != null) {
 				href = HTMLDecoder.decode(href);
-				href = htmlSanitizeURI(href, type, charset, maybecharset, pc.cb, pc, false);
+				if (isIcon) {
+					href = htmlSanitizeURI(href, type, null, null, pc.cb, pc, false);
+				} else {
+					href = htmlSanitizeURI(href, type, charset, maybecharset, pc.cb, pc, false);
+				}
```

Comment:
Why don’t you set the charset and maybecharset to null if (isIcon) and then call htmlSanitizeURI(href, type, charset, maybecharset, pc.cb, pc, false)?